File: LeetCode/0026. Remove Duplicates from Sorted Array/solution.cpp
/**
 * 161 / 161 test cases passed.
 * Runtime: 8 ms
 * Memory Usage: 13.4 MB 
 */
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) return 0;
        int slow = 0;
        for (int fast = 1; fast < nums.size(); ++ fast) {
            if (nums[fast] != nums[slow]) {
                nums[++slow] = nums[fast];
            }
        }
        return slow + 1;
    }
};

File: LeetCode/0026. Remove Duplicates from Sorted Array/solution.py
"""
161 / 161 test cases passed.
Runtime: 32 ms
Memory Usage: 15.6 MB
"""
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0
        size = 0
        for num in nums[1:]:
            if nums[size] != num:
                size += 1
                nums[size] = num
        return size + 1
